src/app/reporting/Engine.cpp

```diff
@@ -176,6 +196,10 @@ DataModel::ActionReturnStatus RetrieveClusterData(DataModel::Provider * dataMode
         // the are routed through metadata.
         status = ReadGlobalAttributeFromMetadata(dataModel, readRequest.path, attributeValueEncoder);
     }
+    else if (auto readable_status = ValidateAttributeIsReadable(dataModel, path); readable_status.has_value())
```

Comment:
Should we move this validation before `ValidateReadAttributeACL()`?

Reply:
I would do this as a followup, maybe part of synchronizing with https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/11812 

My logic here would be: tests now pass, so logic is probably ok at least as far as we test for it. Since this is touchy I would update it as followups only.